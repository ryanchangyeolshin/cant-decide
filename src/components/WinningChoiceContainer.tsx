import React, { useContext } from 'react';
import {
  Grow,
  Typography,
  Button
} from '@material-ui/core';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import { useHistory } from "react-router-dom";
import DecisionContext from '../context/DecisionContext';
import DecisionState from '../types/DecisionState';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    contentContainer: {
      display: "flex",
      flexDirection: "column",
      alignItems: "center",
      justifyContent: "space-between",
      height: "100vh",
      margin: "50px 50px",
      borderRadius: "20px",
      backgroundColor: "#154A40",
      color: "white",
    },
    winningChoice: {
      fontWeight: 900,
      color: "white",
      margin: "50px 50px",
      padding: "0 20px",
      width: "100%",
      wordWrap: "break-word",
      display: "flex",
      justifyContent: "center",
    },
    returnHomeButton: {
      backgroundColor: "#F4ECDA",
      color: "black",
      marginTop: "20px",
      marginBottom: "20px",
      "&:hover": {
        backgroundColor: "yellow"
      },
      "& .MuiButton-label": {
        fontWeight: 900
      }
    }
  }),
);

const WinningChoiceContainer: React.FC = () => {
  const history = useHistory();
  const classes = useStyles();
  const { winningChoice, decision, setDecision } = useContext<DecisionState>(DecisionContext);

  const returnHomeHandler: React.MouseEventHandler<HTMLButtonElement> = (e: React.MouseEvent<HTMLButtonElement>) => {
    e.preventDefault();
    setDecision({ ...decision, choices: [] });
    history.push("/");
  }

  return (
    <Grow in={true} timeout={1500} mountOnEnter unmountOnExit={true}>
      <Typography component="div" className={classes.contentContainer}>
        <Typography variant="h2" className={classes.winningChoice}>
          {winningChoice}
        </Typography>
        <iframe
          title="shia-labeouf-giphy"
          src="https://giphy.com/embed/GcSqyYa2aF8dy"
          width="480"
          height="358"
          frameBorder="0"
          className="giphy-embed"
          allowFullScreen>
        </iframe>
        <Button
          className={classes.returnHomeButton}
          data-testid="randomize-button"
          variant="contained"
          color="primary"
          onClick={returnHomeHandler}
        >
          Return Home
        </Button>
      </Typography>
    </Grow>
  );
}

export default WinningChoiceContainer;