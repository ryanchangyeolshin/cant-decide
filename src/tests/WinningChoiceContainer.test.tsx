import React from 'react';
import { render, cleanup } from '@testing-library/react';
import WinningChoiceContainer from '../components/WinningChoiceContainer';

afterEach(cleanup);

describe("This suit is to test the WinningChoiceContainer component", () => {
  test('Snapshot of WinningChoiceContainer', () => {
    const { asFragment } = render(<WinningChoiceContainer />);
    expect(asFragment()).toMatchSnapshot()
  });

  test('that the RANDOMIZE button is rendered', async () => {
    const { container } = await render(<WinningChoiceContainer />);
    const returnHomeButton = container.querySelector('button') as HTMLElement;
    expect(returnHomeButton.textContent).toBe("Return Home");
  });
});
