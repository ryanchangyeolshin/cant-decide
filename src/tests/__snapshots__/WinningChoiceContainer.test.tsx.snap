// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This suit is to test the WinningChoiceContainer component Snapshot of WinningChoiceContainer 1`] = `
<DocumentFragment>
  <div
    class="MuiTypography-root makeStyles-contentContainer-1 MuiTypography-body1"
    style="opacity: 1; transform: scale(1, 1); transition: opacity 1500ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 999ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <h2
      class="MuiTypography-root makeStyles-winningChoice-2 MuiTypography-h2"
    />
    <iframe
      allowfullscreen=""
      class="giphy-embed"
      frameborder="0"
      height="358"
      src="https://giphy.com/embed/GcSqyYa2aF8dy"
      title="shia-labeouf-giphy"
      width="480"
    />
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-returnHomeButton-3 MuiButton-containedPrimary"
      data-testid="randomize-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Return Home
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
